src/Activities/Internal/ActivityMetricsSender.cs

```diff
@@ -21,70 +18,61 @@ internal sealed class ActivityMetricsSender : IActivitiesEventSender, IDisposabl
 		private readonly Histogram<long> m_healthCheckActivityHistogram;
 		private readonly IExecutionContext m_context;
 		private readonly IHostEnvironment m_hostEnvironment;
-		private readonly ArrayPool<KeyValuePair<string, object?>> m_arrayPool;
+		private readonly HashSet<string> m_extraBaggageDimension;
+		private readonly HashSet<string> m_extraTagObjectsDimension;
 
-		public ActivityMetricsSender(IExecutionContext executionContext, IHostEnvironment hostEnvironment)
+		public ActivityMetricsSender(IExecutionContext executionContext, IHostEnvironment hostEnvironment, IExtraActivityBaggageDimensions extraActivityBaggageDimensions, IExtraActivityTagObjectsDimensions extraActivityTagObjectsDimensions)
 		{
 			m_context = executionContext;
 			m_hostEnvironment = hostEnvironment;
 			m_meter = new Meter("Microsoft.Omex.Activities", "1.0.0");
 			m_activityHistogram = m_meter.CreateHistogram<long>("Activities");
 			m_healthCheckActivityHistogram = m_meter.CreateHistogram<long>("HealthCheckActivities");
-			m_arrayPool = ArrayPool<KeyValuePair<string, object?>>.Create();
+			m_extraBaggageDimension = extraActivityBaggageDimensions.ExtraDimensions;
+			m_extraTagObjectsDimension = extraActivityTagObjectsDimensions.ExtraDimensions;
 		}
 
 		public void SendActivityMetric(Activity activity)
 		{
-			long durationMs = Convert.ToInt64(activity.Duration.TotalMilliseconds);
-
-			int tagsCount = s_customTags.Length + activity.TagObjects.Count() + activity.Baggage.Count();
-
-			KeyValuePair<string, object?>[] tags = m_arrayPool.Rent(tagsCount);
-
-			int index = 0;
+			Histogram<long> histogram = activity.IsHealthCheck() ? m_healthCheckActivityHistogram : m_activityHistogram;
 
-			foreach (Func<ActivityMetricsSender, Activity, KeyValuePair<string, object?>> getter in s_customTags)
-			{
-				tags[index++] = getter(this, activity);
-			}
+			long durationMs = Convert.ToInt64(activity.Duration.TotalMilliseconds);
 
-			foreach (KeyValuePair<string, string?> baggage in activity.Baggage)
+			TagList tagList = new()
+				{
+					{ "Name", activity.OperationName },
+					{ "RegionName", m_context.RegionName },
+					{ "ServiceName", m_context.ServiceName },
+					{ "BuildVersion", m_context.BuildVersion },
+					{ "Environment", m_hostEnvironment.EnvironmentName },
+					{ "Cluster", m_context.Cluster },
+					{ "ApplicationName", m_context.ApplicationName },
+					{ "NodeName", m_context.NodeName },
+					{ "MachineId", m_context.MachineId },
+					{ "DeploymentSlice", m_context.DeploymentSlice },
+					{ "IsCanary", m_context.IsCanary },
+					{ "IsPrivateDeployment", m_context.IsPrivateDeployment }
+				};
```

Comment:
From documentation:
When emitting metrics with more than eight tags, the SDK allocates memory on the hot-path. You SHOULD try to keep the number of tags less than or equal to eight. Check if you can extract any shared tags such as MachineName, Environment etc. into Resource attributes.

Reply:
I have been keeping this in mind. I can do another PR to reduce unnecessary dimensions such as Environment